src/construct.ts

```diff
@@ -417,21 +422,13 @@ export class Node {
       throw new Error(`Cannot add children to "${this.path}" during synthesis`);
     }
 
-    if (childName in this._children) {
```

Comment:
From what I tested, array access is slightly faster than using the "in" operator